sdk/inc/azure/core/internal/az_spinlock_internal.h

```diff
@@ -4,7 +4,7 @@
 #ifndef _az_SPINLOCK_INTERNAL_H
 #define _az_SPINLOCK_INTERNAL_H
 
-#include <azure/core/_az_spinlock.h>
+#include <azure/core/az_credentials.h>
```

Comment:
Hmm, I think we may want to revisit the move of spinlock to az_credential.h. I didn't realize it was used in internal headers like this.

Now, this seems odd, having spinlock seemingly rely on credentials.

Reply:
I knew we run into this, and didn't see an issue in this. Let's ask @JeffreyRichter?

Reply:
It's fine to keep the _az_spinlock.h public header file.

Reply:
Hmm, ok, I'll undo this change.